src/hotspot/share/prims/jvmtiThreadState.cpp

```diff
@@ -557,7 +557,9 @@ JvmtiVTMSTransitionDisabler::VTMS_vthread_start(jobject vthread) {
   // If interp_only_mode has been enabled then we must eagerly create JvmtiThreadState
   // objects for globally enabled virtual thread filtered events. Otherwise,
   // it is an important optimization to create JvmtiThreadState objects lazily.
-  if (JvmtiThreadState::seen_interp_only_mode()) {
+  if (JvmtiThreadState::seen_interp_only_mode() ||
+      JvmtiExport::should_post_field_access() ||
+      JvmtiExport::should_post_field_modification()){
```

Comment:
The comment needs updating to explain the extra checks.

Can't say I see the connection with [8316283](https://bugs.openjdk.org/browse/JDK-8316283) as no `-Xcomp` is involved ion the current failures AFAICS.

Reply:
Thank you for the suggestion. I'll add a comment.
The `-Xcomp` flag is not a root cause but a trigger. There is a general issue that a frame can be not deoptimized. The `-Xcomp` is a stress that helps to reproduce the issue as the `-Xcomp` option and `interp_only_mode` works in opposite directions.